doc/compiler/optimizer/OptimizerDesignFeatures.md

```diff
@@ -0,0 +1,303 @@
+
+Of course, the OpenJ9 JIT compiler does have an OSR infrastructure now as well, so the choice on what
+to do on fall-back paths can be chosen heuristically.
+
+## Avoidance of Expensive Analyses
+
+Despite being selective with compiling methods, using lower optimization levels and hot path information to
+reduce compile time, it is still a reality that some analyses that may be considered essential in a static
+compiler are simply too expensive in a JIT compiler.
+
+For example, the OpenJ9 JIT compiler analyses such as value numbering have been implemented to function
+without Static Single Assignment(SSA) form representation, which would be too expensive to compute and
+maintain/recompute as optimizations transform the IL of the compiled method.
+
+SSA in particular would make the task of implementing certain analyses and optimizations simpler but the
+extra complexity in implementation effort was a trade-off that was worth the compile time savings that
+accrued as result.
+
+Another example of a standard static compiler analysis not being present in the OpenJ9 JIT compiler is
+sophisticated aliasing of any sort involving pointers or even much code analysis.
+
+While presenting the lack of sophisticated aliasing as a feature may appear counter-intuitive at first, it
+makes sense for a JIT compiler given the subtlety that JIT compilers are typically employed in languages with
+a high degree of dynamism both in terms of call targets and the side effects that are permitted at various
+program points.
+
+Thus, the payoff of doing a sophisticated aliasing analysis is likely not worth the compile time cost and
+the basic aliasing that is there relies more on properties of the fundamental
+properties of the symbols in use, e.g. a call aliases all static symbols.
+
+Of course, the aliasing in particular can be extended or improved fairly easily and the optimizations in
+the compiler are shielded from the internals of how the aliasing is computed
+via well-defined APIs.
+
+Interprocedural analysis is another concept that is common in static compilers but once again the payoffs
+in a dynamic language for a JIT compiler are probably not worth the effort to build and maintain a call
+graph and associated data structures.
+
+Another cool capability in the OpenJ9 JIT compiler that somewhat softens the blow of having to run
+optimizations iteratively is the capability that optimizations must enable other optimizations as they run.
+In other words, an optimization can either create or detect an opportunity that makes it important to run
+another optimization, but the infrastructure allows later optimization passes to be enabled on-demand rather
+having to be run unconditionally, meaning we only pay the compile time of repeating optimizations if there
+is some expected benefit.
+
+## IL Representation
+
+The OpenJ9 JIT compiler uses a (Directed Acyclic Graph) DAG to represent individual computations in
```

Comment:
A link to [Testarossa's Intermediate Language: An Intro to Trees](https://github.com/eclipse/omr/blob/master/doc/compiler/il/IntroToTrees.md) might be appropriate here.

Reply:
I'll add the hyperlink down to where "IL representation" is mentioned again. There is a bookmark in the beginning of the article to jump to this section using `#il-representation`. Not sure what format to use if the section title has a a hyperlink